controller.go

```diff
@@ -426,11 +426,15 @@ func (c *controller) RegisterDriver(networkType string, driver driverapi.Driver,
 
 // NewNetwork creates a new network of the specified network type. The options
 // are network specific and modeled in a generic way.
-func (c *controller) NewNetwork(networkType, name string, options ...NetworkOption) (Network, error) {
+func (c *controller) NewNetwork(networkType, name string, id string, options ...NetworkOption) (Network, error) {
 	if !config.IsValidName(name) {
 		return nil, ErrInvalidName(name)
 	}
 
+	if id == "" {
+		id = stringid.GenerateRandomID()
```

Comment:
If you generate the id here, then don't do it again when you initialize the network structure.